test/e2e2/all_in_one_test.go

```diff
@@ -0,0 +1,50 @@
+package e2e2_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e/cli"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e2/kube"
+)
+
+var _ = Describe("all-in-one.yaml", Ordered, Label("all-in-one"), func() {
+	var kubeClient client.Client
+
+	_ = BeforeAll(func() {
+		ctx := context.Background()
+		c, err := kube.NewK8sTest(ctx)
+		kubeClient = c
+		Expect(err).To(Succeed())
+	})
+
+	It("applies all-in-one.yaml", func() {
+		Expect(cli.Execute("kubectl", "apply", "-f", "../../deploy/all-in-one.yaml").Wait().ExitCode()).Should(Equal(0))
+	})
+
+	It("waits for mongodb-atlas-operator deployment to be Ready", func() {
```

Comment:
nit: I see these as part of the same test (one It), but as this is ordered, I guess the behaviour will be the same.